3924 mod archive thread

Miaplacidus opened the pull request:
Provide ability for mods/admins to archive threads.

## Link to Issue
Closes: #3924 

## Description of Changes
- Allows mods/admins to archive threads. 
- Conditionally disables/hides elements based on archive status

## Test Plan
- inspect thread index
- inspect view thread page

## Deployment
- should be tested against changes in #4190, which includes the backend updates for this to work for the user
-  should be deployed *after* #4190 is merged in 

## Other Considerations
- will push to demo branch for viewing once #4121 has been approved

jnaviask commented on packages/commonwealth/.storybook/stories/atoms/Button.stories.tsx:
```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import type { Meta, StoryObj } from "@storybook/react";
+
+<<<<<<< HEAD:packages/commonwealth/.storybook/stories/old/atoms/Button.stories.tsx
```
Merge conflict need resolving?

Miaplacidus replied:
Just resolved!

Miaplacidus commented:
@rbennettcw Merged in your changes! Thanks! 
It is missing one thing - we'll need to handle navigation to the page for threads that are archived.

jnaviask commented:
@Miaplacidus please mark as draft until this PR is ready for final review + merging. Thanks!

masvelio commented:
@Miaplacidus how can we make this mergable any time soon?

jnaviask commented:
cc @Miaplacidus to fix merge conflicts then we can do a review

zakhap commented:
https://github.com/hicommonwealth/commonwealth/assets/16248081/cb947da9-c5f4-4ab5-b9d6-8bfd5d45cce5

After archiving a thread, if I click the "include archived posts" checkbox, they don't appear again. Also a weird glitch when I went to the archive page where the threads didn't show up? They do sometimes tho

jnaviask commented:
Going to mark as draft -- please return to ready for review once merge conflicts + bugs are resolved. cc @Miaplacidus

Miaplacidus commented:
@jnaviask @zakhap 
Merge conflicts sorted and bugs resolved. 
The failing Quality Check keeps referring to a file that no longer exists. Not sure why.

Miaplacidus commented:
@zakhap @jnaviask @CowMuon 
I've looked everywhere for references to the file the CI claims is missing. Perhaps there's a caching issue...? Having said that, I have noticed some bugginess in the Code Quality Checker in general. E.G. it will note that you are importing something which you don't use later in the file...even when it is definitely being used later in the file.

jnaviask commented:
Looks like @kurtisassad work in #4770 wasn't merged properly? Or there's some confusion around the New Thread Tag? Please pair when you get a chance to figure out what happened.

Miaplacidus commented:
@jnaviask
@kurtisassad just pushed up a fix! Thanks again, Kurtis!

jnaviask commented on packages/commonwealth/client/scripts/state/api/threads/toggleArchive.ts:
```diff
@@ -14,15 +15,19 @@ const toggleThreadArchive = async ({
       } as any,
+    },
+    {
+      headers: {
```
We don't use JWT headers -- providing it as part of the data alone is sufficient.

Miaplacidus replied:
> We don't use JWT headers -- providing it as part of the data alone is sufficient.

During the resolution of the merge topics, I noticed that some API logic had been updated. I attempted to just pass the JWT as part of the data (which is what I originally did), but that resulted in repeated "Unauthorized" errors. Placing the JWT in the header fixed the issue. Perhaps there is some aspect of the backend logic that will require an update? @rbennettcw

rbennettcw replied:
You can pass the token as the `jwt` property in the request body or as the bearer token in the headers. Both should work.

rbennettcw replied:
BTW, with the latest code, you can/should use the `PATCH /threads/:id` route. It supports archive/unarchive.

The request body would be `{ archive: boolean }`, you can set to `true` or `false` and it's returned as a timestamp or `null`.

For the patch route, you may get the `Unauthorized` error if you don't have permissions to perform the operation.

Miaplacidus replied:
Updated code to use the new thread editing logic. Thanks!

jnaviask commented on packages/commonwealth/client/scripts/models/Thread.ts:
Noticing a bunch of code quality checks in files touched -- can these be resolved?

Miaplacidus replied:
Yep yep! Resolved!

jnaviask commented on packages/commonwealth/client/scripts/views/components/Comments/CreateComment.tsx:
```diff
@@ -158,55 +159,73 @@ export const CreateComment = ({
-      </div>
-    </div>
+    <>
+      { rootThread.archivedAt === null ? (
```
Can we split this out into separate components, following the same pattern as in #4776, rather than using a big ternary? We can do an `if (archived) return ArchivedComponent; return MainComponent;` pattern to make this more readable.

jnaviask commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWUpvoteSmall.tsx:
```diff
@@ -44,7 +46,8 @@ const CWUpvoteSmall = ({
             >
               <CWThreadAction
-                action="upvote"
+                action= "upvote"
```
Spacing

masvelio replied:
this should not be a case if prettier is correctly steup, but it seems that it is not on your machine @Miaplacidus ?

https://commonxyz.slack.com/archives/C050AE0URFH/p1689075560213459

jnaviask commented on packages/commonwealth/client/scripts/views/components/sidebar/discussion_section.tsx:
```diff
@@ -43,6 +44,10 @@ export const DiscussionSection = () => {
     location
   );
+  const matchesArchivedRoute = matchRoutes(
+    [{ path: '/discussions/Archived' }, { path: ':scope/discussions/Archived' }],
```
We shouldn't use casing for paths.

jnaviask commented on packages/commonwealth/client/scripts/views/components/sidebar/discussion_section.tsx:
```diff
@@ -146,6 +151,28 @@ export const DiscussionSection = () => {
+      rightIcon: <CWIcon iconName="archiveTray" iconSize="small" />,
+      containsChildren: false,
+      hasDefaultToggle: false,
+      isVisible: true,
```
cc @zakhap do we always want "Archived" visible? Or only if threads have been archived?

jnaviask commented on packages/commonwealth/client/scripts/views/components/sidebar/sidebar_section.tsx:
```diff
@@ -138,7 +138,8 @@ const SubSectionGroup = (props: SectionGroupAttrs) => {
           {title}
         </CWText>
-        {rightIcon && <div className="right-icon">{rightIcon}</div>}
+        {/* {rightIcon && <div className="right-icon">{rightIcon}</div>} */}
```
Please remove dead code

masvelio replied:
remove dead code

kurtisassad commented:
Seems that some bugs that the CI caught correspond to real breakages. If you figure out how to fix this issue on the discussions page let me know, and I will manually recheck CI to make sure that they aren't flaky failures.
<img width="562" alt="Screenshot 2023-08-16 at 8 52 49 AM" src="https://github.com/hicommonwealth/commonwealth/assets/14794654/67965e7d-c1f3-4df6-9a64-02aa252d079c">

jnaviask commented on packages/commonwealth/client/scripts/views/pages/discussions/CommentCard/CommentCard.tsx:
```diff
@@ -40,6 +40,7 @@ type CommentCardProps = {
   canToggleSpam?: boolean;
   // actual comment
   comment: Comment<any>;
+  threadArchivedAt: moment.Moment | null;
```
Any reason we need the full date rather than a boolean for this component?

jnaviask reviewed (changes requested):
Beyond the inline comments, there's one very significant consideration here: we have added a "reserved topic name". What would happen wrt this PR if I made my own custom topic named "Archived"?

Two options to resolve:
1. Continue with the reserved topic name, make server-side changes forbidding users from setting topics named "Archive".
2. Add top-level client route to `/:community/archived` (or just `/archived` on custom domains), which displays the discussions page with a flag to only show archived threads.

I think option 2 is best, in that reserved namespaces is a dangerous game, despite the fact that it will require more refactoring work. Does this make sense @Miaplacidus ?

Miaplacidus commented:
> Seems that some bugs that the CI caught correspond to real breakages. If you figure out how to fix this issue on the discussions page let me know, and I will manually recheck e2e tests to make sure that they aren't flaky failures. <img alt="Screenshot 2023-08-16 at 8 52 49 AM" width="562" src="https://user-images.githubusercontent.com/14794654/261064663-67965e7d-c1f3-4df6-9a64-02aa252d079c.png">

Hey, @kurtisassad! Could you describe the actions you took to generate that error?

kurtisassad commented:
Sorry Ifu. It was a false alarm. This bug was actually introduced in a previous commit in master but has been fixed. So provided that you merged in the latest master, this should disappear.

For reference this page would show up on the /discussions or /discussion pages

Miaplacidus commented:
> Beyond the inline comments, there's one very significant consideration here: we have added a "reserved topic name". What would happen wrt this PR if I made my own custom topic named "Archived"?
> 
> Two options to resolve:
> 
> 1. Continue with the reserved topic name, make server-side changes forbidding users from setting topics named "Archive".
> 2. Add top-level client route to `/:community/archived` (or just `/archived` on custom domains), which displays the discussions page with a flag to only show archived threads.
> 
> I think option 2 is best, in that reserved namespaces is a dangerous game, despite the fact that it will require more refactoring work. Does this make sense @Miaplacidus ?

I agree. Archived posts are not really a topic, yet we're treating them as kinda-sorta such. We'll need a card for the refactor of the backend logic. @rbennettcw should be pinged on this.

rbennettcw commented:
> > Beyond the inline comments, there's one very significant consideration here: we have added a "reserved topic name". What would happen wrt this PR if I made my own custom topic named "Archived"?
> > Two options to resolve:
> > 
> > 1. Continue with the reserved topic name, make server-side changes forbidding users from setting topics named "Archive".
> > 2. Add top-level client route to `/:community/archived` (or just `/archived` on custom domains), which displays the discussions page with a flag to only show archived threads.
> > 
> > I think option 2 is best, in that reserved namespaces is a dangerous game, despite the fact that it will require more refactoring work. Does this make sense @Miaplacidus ?
> 
> I agree. Archived posts are not really a topic, yet we're treating them as kinda-sorta such. We'll need a card for the refactor of the backend logic. @rbennettcw should be pinged on this.

Is `Continue with the reserved topic name, make server-side changes forbidding users from setting topics named "Archive".` the only change needed?

Miaplacidus commented:
> > > Beyond the inline comments, there's one very significant consideration here: we have added a "reserved topic name". What would happen wrt this PR if I made my own custom topic named "Archived"?
> > > Two options to resolve:
> > > 
> > > 1. Continue with the reserved topic name, make server-side changes forbidding users from setting topics named "Archive".
> > > 2. Add top-level client route to `/:community/archived` (or just `/archived` on custom domains), which displays the discussions page with a flag to only show archived threads.
> > > 
> > > I think option 2 is best, in that reserved namespaces is a dangerous game, despite the fact that it will require more refactoring work. Does this make sense @Miaplacidus ?
> > 
> > 
> > I agree. Archived posts are not really a topic, yet we're treating them as kinda-sorta such. We'll need a card for the refactor of the backend logic. @rbennettcw should be pinged on this.
> 
> Is `Continue with the reserved topic name, make server-side changes forbidding users from setting topics named "Archive".` the only change needed?

Hey, @rbennettcw. We're going with option number 2.

jnaviask commented:
I'm not sure why @rbennettcw needs to make any changes? I'm speaking specifically about the client-side routing. The server should not be affected at all.

Miaplacidus commented:
@jnaviask I've addressed the prior comments (including the code quality issues) in the most recent commits. I'm not sure if the most recent push to demo resulted in a deployment, however.

Miaplacidus commented:
Never mind! It's been deployed!

jnaviask commented on packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx:
```diff
@@ -339,14 +349,25 @@ export const HeaderWithFilters = ({
+          }}
+        />
+
+        { !onArchivePage &&
```
Linter fix here.

jnaviask reviewed (commented):
Code looks good on quick glance through. Will want to do some playing around with the UI as final pass to verify, but we may be good to go here.

jessmart1213 commented:
### Design QA Feedback ###

@Miaplacidus  Overall the archive functionality looks good! I highlighted below some minor updates and some things that were missing. Lmk if you anything needs more clarification! 

---

**Archive list item placement**

<img width="610" alt="Screen Shot 2023-08-28 at 4 36 27 PM" src="https://github.com/hicommonwealth/commonwealth/assets/133259695/7e538a6e-c670-4f64-9106-81b38cdbd72a">

- Note how Archived is sitting above the General topic list item. 
- Update Archived list item to be the last (bottom-most) list item within the Discussion section for all community members and every community.

**Archived page header copy**

<img width="796" alt="Screen Shot 2023-08-28 at 4 45 05 PM" src="https://github.com/hicommonwealth/commonwealth/assets/133259695/04d90aed-5391-4aa4-8604-738e74e275c5">

- Descriptive copy underneath Header is missing. 
- Please update to include to copy -> "This section is for all archived posts. Archived posts will always be visible here and can be linked to new thread posts, but they can’t be upvoted or receive new comments."

**Archived page empty state copy**

- When an Archived page has no archived threads update current copy to ->
"There are currently no archived threads."
- ^Was not specified in original issue, so this on Product & Design,  but update seems low effort. Lmk if a new ticket is needed for this.

**More actions dropdown menu**

<img width="745" alt="Screen Shot 2023-08-28 at 4 54 26 PM" src="https://github.com/hicommonwealth/commonwealth/assets/133259695/2fe3b5b7-0b97-428d-80bc-1516d5b2f48d">

- Update the dropdown menu from default to compact so that list items have a height of 32px instead of 40px. Update on the thread page and on the thread listing row component.

**Modal copy size**
- Update modal copy from Body 01/Regular to Body 02/Regular to match designs.
- I recognize that updates to this modal may be done when Daniel's design system modal work updates modals across the site, so will leave up to Eng to reconcile. 

**Archived Icon**

<img width="839" alt="Screen Shot 2023-08-28 at 5 07 16 PM" src="https://github.com/hicommonwealth/commonwealth/assets/133259695/a921579c-c1e9-481b-b259-bd33b8880d69">

- Update archived thread pages to show the filled Archive icon within Metadata as seen in designs.
- Ensure that the archive icon has tooltip that is triggered on:hover as shown in design mocks.
- The Archive icon should be behave as a Locked thread icons do, appearing on thread listing rows as well as on the thread page. This is especially important when users select the "Show archived threads" checkbox, as they should currently have no way to identify archived threads from none archived threads. However, this was not specified in the designs or the issue, so it will be ticketed as a fast follow.

CowMuon commented:
It looks like this PR was not set back to draft after Jessica's design feedback last week. 

PR's that are not ready for review (meaning codeowners) should be set to draft (which is generally, not necessarily strictly, done by codeowner reviewing, but is also subject to automation.) 

In the case of a Design Review (ie. not the same as Codeowner Review), feedback that isn't purely 👍 should result in status changing back to draft for remaining work. 

You can see that this PR  is _**ambiguous**_ as to whether it's ready for review (given Jess' comments, subsequent commits by Ifu, and the unchanged status of "Open") and ofc such ambiguity is exactly what we strive to eliminate. @Miaplacidus

cc. @gdjohnson so make sure our documentation isn't, er, ambiguous on this point.

Miaplacidus commented:
@jessmart1213 
Thanks for the detailed feedback, Jess! 
I've addressed each of your concerns as follows:
- moved the 'Archived' link to the bottom of the Discussion section of the sidebar
- provided copy has been placed underneath the 'Archived' header
- provided copy has been used when no archived threads are present
- have not updated the dropdown menu to be more compact. Will create a card to address dropdown menu sizing
- updated the modal text (specifically, the line height, which was the only difference I saw)
- the filled archive tray icon has been added to archived threads along with a tooltip that appears when the icon is hovered over. However, the tooltip has the original tooltip design (i.e. without an arrow) since we have new tooltip designs that we have yet to insert into the app. We'll need to create a card/multiple cards around tooltip implementation.

CowMuon commented:
@jessmart1213 If you can sign off on Design Review, we can advance this to Under (codeowner) Review on this side. Thanks, this has been a long train (literally since June 26 🙀) and hopefully the _last_ of the long trains!

Miaplacidus commented:
@martins87 @masvelio @jnaviask @MuonShot 
Hey, guys! 
I'd like to get this out the door today. I've addressed the feedback provided by Jessica earlier (see my most recent comment above).

What I need checked:
* that the Archived link is now at the bottom of the Discussions section of the sidebar
* the archived page now has copy indicating what it's for
* when there are no archived posts, the archived page displays copy saying as much
* archived posts have a little filled in archive tray and tooltip describing when the post was archived
* the line height of text in the archive modal looks good

jessmart1213 commented:
Hi @Miaplacidus! 

I only have one piece of feedback, but everything else looks GREAT! Thank you so much! 

---

Not quite sure what was changed with the line height specifically, but when I compare the implemented modal copy size to the designs I still still see a size difference.

<img width="856" alt="Screen Shot 2023-09-11 at 6 35 43 PM" src="https://github.com/hicommonwealth/commonwealth/assets/133259695/60159601-aeff-46cb-b7f4-babd3284996b">

Update modal copy from Body 01/Regular to Body 02/Regular. Again, Daniel's modal work may fix this so up to you and Eng on how to tackle. Lmk!

CowMuon commented:
JTN if this is not blocking Session Keys, it is backseat priority, as Product Design only has today for Design Review sign off (basically Jess as only designer; sign off can come from Sachin or Zak, ideally both.) 

That being said, the PR description does need to be improved, here, but moreover going forward; to wit: 
- Description of Changes "Allows mods/admins to archive threads" literally just repeats the issue description. Rather, this section should describe _what was done in order to allow that_
- What is the difference here between "disabling" and "hiding" elements based on archive status?
- Test Plan is not really sufficient here. Keep in mind our 2 test plan description levels: 
    0.  Description is sufficient for a developer with a lot of contextual knowledge.
    1.  Description is sufficient for a developer without extensive contextual knowledge of this area.
    2. Description is considered sufficient for our "developer api endpoint" (read: GPT) to take a pass at it. 
#0 (zero) here is right out, ofc. #1 is minimum requirement ("must" using our preferred RFC 2119 format) and #2 is the stretch goal ("should" in RFC 2119).

CowMuon commented:
@gdjohnson Making sure charter includes RFC 2119 definitions for both new issue descriptions and PR test plans is a TOP priority. Let's discuss on 9/12.

Miaplacidus commented:
@jessmart1213 @MuonShot 
I'm guessing Daniel's modal work addresses the new stylings specific to modals. I'm happy to make card to replace this modal with his new work and put it on my plate immediately, but it would be nice to get this card merged in. It contains a lot of work and, frankly, further delays means further merge conflicts.

jessmart1213 commented:
@Miaplacidus Im happy with your suggestion for handling the modal copy. So giving this PR, Design sign-off.

masvelio commented:
@Miaplacidus for visibility, Jessica found some archive related issue on demo env https://github.com/hicommonwealth/commonwealth/pull/5138#issuecomment-1737386437

Miaplacidus commented:
@jnaviask @MuonShot 
Just resolved the latest conflicts. Should be good to go!

jnaviask commented:
cc @Miaplacidus please fix code quality CI issues

Miaplacidus commented:
@jnaviask Seems like the missing parens was introduced with the merge from master, but it's fixed now. Also included some formatting improvements.

CowMuon commented:
@Miaplacidus 38 files is far too many to be including as formatting changes in the same commit that has your logical changes that directly resolve the issue. We need to know whether you were not aware of our code style on this (in which case we need to disseminate better) or just wasn't thinking about that (in which case we'll assume documentation doesn't need to be updated.)

Miaplacidus commented:
@CowMuon I believe the reference to 38 files being changed is the total number of files changed on the branch; not the changes I just made. My most recent commit touched only 3 files.

Miaplacidus commented:
@jnaviask What's the timeline on getting this merged?

Miaplacidus commented:
@masvelio @jessmart1213 @rbennettcw 
If this feature seems good to go, would a couple of you mind approving this so it can be merged?

jessmart1213 commented:
@Miaplacidus In case this needs another design sign-off, here it is - Design sign-off!

masvelio commented on packages/commonwealth/client/scripts/state/api/threads/fetchThreads.ts:
```diff
@@ -186,7 +188,7 @@ const useFetchThreadsQuery = (
       if (isFetchBulkThreadsProps(props)) {
         return {
-          getNextPageParam: (lastPage, pages) => lastPage.pageParam,
+          getNextPageParam: (lastPage, _pages) => lastPage.pageParam,
```
'_pages' is defined but never used

Miaplacidus replied:
Removed!

masvelio commented on packages/commonwealth/client/scripts/views/components/archive_tray_with_tooltip.tsx:
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { CWTooltip } from './component_kit/cw_popover/cw_tooltip';
```
we should not use old tooltip anymore. We should use one from `new_designs`

Miaplacidus replied:
Done!

masvelio commented on packages/commonwealth/client/scripts/views/components/Comments/CommentEditor/CommentEditor.tsx:
```diff
@@ -0,0 +1,112 @@
+  const decimals = getDecimals(app.chain);
+
+  return (
+    <div className="CreateComment">
```
this is great that you decided to move comment editor to separate file 💪 but if we do this, we should remember about the styles as well and general rules we have for creating files/componnets/namings etc. So in this case:
- main class name should be changed from `CreateComment` to `CommentEditor`
- new scss file `CommentEditor.scss` should be created that would handle only styles for this particular component

Those steps are important to have files and codebase tidy up.

Miaplacidus replied:
Done!

masvelio commented on packages/commonwealth/client/scripts/views/components/archive_tray_with_tooltip.tsx:
```diff
@@ -0,0 +1,28 @@
+      )}
+    />
+  );
+};
```
change name of the file archive_tray_with_tooltip.tsx => ArchiveTrayWithTooltip.tsx

Miaplacidus replied:
Done!

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_thread_action.tsx:
```diff
@@ -110,7 +111,11 @@ export const TooltipWrapper = ({
 
-const getTooltipCopy = (action: ActionType) => {
+const getTooltipCopy = (action: ActionType, threadArchived: boolean) => {
+  if (threadArchived) {
+    return 'Thread is archived';
+  }
```
![image](https://github.com/hicommonwealth/commonwealth/assets/14819225/1df028d7-486f-4bce-a281-28e397f2021f)

This does not seem to be working 🤔

Miaplacidus replied:
Fixed!

masvelio commented on packages/commonwealth/client/scripts/views/components/Comments/ArchiveMsg/ArchiveMsg.tsx:
```diff
@@ -0,0 +1,24 @@
+      </div>
+    </div>
+  )
+}
```
prettier: file is not formatted correctly

Miaplacidus replied:
Done!

masvelio commented on packages/commonwealth/client/scripts/views/components/Comments/CommentEditor/CommentEditor.tsx:
```diff
@@ -0,0 +1,112 @@
+      </div>
+    </div>
+  )
+}
```
prettier: file is not formatted correctly

Miaplacidus replied:
Done

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/cw_icons/cw_icons.tsx:
```diff
@@ -1233,6 +1293,36 @@ export const CWDots = (props: IconProps) => {
+    </svg>
+  );
+};
+
```
1. prettier: file is not formatted correctly
2. Please do not use properties with dashes. They should all be transformed to camelCase properties. Otherwise, we have a lot of warnings and noise in the console. So `stroke-width` => `strokeWidth` etc

masvelio commented on packages/commonwealth/client/scripts/views/components/sidebar/discussion_section.tsx:
```diff
@@ -220,6 +225,31 @@ export const DiscussionSection = () => {
+      );
+    },
+    displayData: null,
+  }
```
prettier: file is not formatted correctly

Miaplacidus replied:
Done!

masvelio commented on packages/commonwealth/client/scripts/views/modals/archive_thread_modal.tsx:
```diff
@@ -0,0 +1,87 @@
+  thread: Thread;
+};
+
+export const ArchiveThreadModal = ({
```
change name of the file `archive_thread_modal.tsx` => `ArchiveThreadModal.tsx`

Miaplacidus replied:
Done!

masvelio commented on packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:
```diff
@@ -80,6 +84,10 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
 
           if (!includeSpamThreads && thread.markedAsSpamAt) return null;
 
+          if (!onArchivePage && !includeArchivedThreads && thread.archivedAt !== null) return null;
```
prettier: file is not formatted correctly

masvelio commented on packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx:
```diff
@@ -141,7 +147,11 @@ export const HeaderWithFilters = ({
-          {isUndefined(topic) ? 'All Discussions' : topic}
+          {
+            isUndefined(topic)
+              ? ( onArchivePage ? 'Archived' : 'All Discussions')
```
prettier: file is not formatted correctly

Miaplacidus replied:
Fixed!

masvelio commented on packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadOptions/AdminActions/AdminActions.tsx:
```diff
@@ -256,6 +258,25 @@ export const AdminActions = ({
 
+  const handleArchiveThread = () => {
+    if (thread.archivedAt === null) {
+      setIsArchiveThreadModalOpen(true)
```
prettier: file is not formatted correctly

masvelio commented on packages/commonwealth/client/scripts/views/pages/overview/TopicSummaryRow.tsx:
```diff
@@ -113,7 +112,7 @@ export const TopicSummaryRow = ({
                       {moment(getLastUpdated(thread)).format('l')}
                     </CWText>
-                    <NewThreadTag threadCreatedAt={thread.createdAt} />
+                    { isNewThread(thread.createdAt) && <CWTag label={'New'} type={'new'} iconName={'newStar'}/> }
```
prettier: file is not formatted correctly

Miaplacidus replied:
Fixed!

masvelio commented on packages/commonwealth/client/styles/modals/archive_thread_modal.scss:
```diff
@@ -0,0 +1,37 @@
+    }
+  }
+
+}
```
1. prettier: file is not formatted correctly
2. Change name of the file to ArchiveThreadModal.scss

Miaplacidus replied:
Done!

masvelio reviewed (approved):
I clicked around and generally it works good. I have some comments to the code itself, but this is more about the style, file names etc. So it should be quick to resolve those.

Prettier on your PRs is still a problem.

I know this PR is up for weeks that is why I am gonna approve in advance, but please fix my requests in the meantime and make sure that you prettier config is correct and it works locally.

Miaplacidus commented:
Bug: Following most recent updates, archived threads no longer immediately removed from thread listing page. Working on fix...

jnaviask commented:
Flagging @mzparacha for final code review

Miaplacidus commented:
@jnaviask @ForestMars 
The archived post disappearance issue is rather flaky. Almost all of the time, the thread disappears from the page when you confirm that you want to archive or unarchive it. Occasionally, it will not, but it's not clear why. I'm guessing something changed in the way we handle state. Would you have any recommendations as to who best to speak to about this?

mzparacha commented on packages/commonwealth/client/scripts/state/api/threads/fetchThreads.ts:
```diff
@@ -35,6 +35,7 @@ interface FetchBulkThreadsProps extends CommonProps {
   topicId?: number;
   stage?: string;
   includePinnedThreads?: boolean;
+  onArchivePage?: boolean;
```
I suggest renaming it to something bool'ish => `is{_something_}` , `should{__something}` =>  `isArchived` etc

NAB: the `includePinnedThreads` could be renamed, I think I added that originally 😅

Miaplacidus replied:
Done!

mzparacha commented on packages/commonwealth/client/scripts/views/components/Comments/ArchiveMsg/ArchiveMsg.tsx:
```diff
@@ -0,0 +1,22 @@
+export const ArchiveMsg = ({ archivedAt }: ArchiveMsgProps) => {
+  return (
+    <div className="ArchiveMsg">
+      <div className="archive-msg-container">
+        <CWIcon iconName="archiveTrayFilled" iconSize="small" />
```
I suggest we use a single format (snake/kebab/camel etc) for classnames

Miaplacidus replied:
For now, icon names are camel-cased. We'll likely stick to this for the foreseeable future.

mzparacha commented on packages/commonwealth/client/scripts/views/components/Comments/CommentEditor/CommentEditor.tsx:
```diff
@@ -0,0 +1,112 @@
+  activeTopic: Topic;
+  userBalance: BN;
+  disabled: boolean;
+  cancel: (e: any) => void;
```
- `cancel => onCancel`
- i suggest we don't pass the react state callback in any component. prop update `setContentDelta => onContentDeltaChange` and in the callback in parent component we set local/react state
- `handleSubmitComment => onSubmitComment / onComment`

Miaplacidus replied:
`cancel` prop changed to `onCancel`.  `cancel` function changed to `handleCancel`.
I'm passing `setContentDelta` down to the Quill editor, which I'm not looking to modify.
I typically prefer using 'handle' for the functions and 'on' for props.

mzparacha commented on packages/commonwealth/client/scripts/views/components/Comments/CommentEditor/CommentEditor.tsx:
```diff
@@ -0,0 +1,112 @@
+          <CWText
+            type="caption"
+            fontWeight="medium"
+            className={clsx('user-link-text', { disabled: !canComment })}
```
`clsx => getClasses`

Miaplacidus replied:
I'm not sure why that's preferable, tbh. Open to changing it, though.

mzparacha commented on packages/commonwealth/client/scripts/views/components/Comments/CommentEditor/CommentEditor.tsx:
```diff
@@ -0,0 +1,112 @@
+import './CommentEditor.scss';
+
+type CommentEditorProps = {
+  parentType: ContentType;
```
can we change `parentType` to be more explicit like and an enum `mode: 'comment' | 'reply'` or a bool `isCommenting: boolean`.

I see we only use it once like this

```
{parentType === ContentType.Comment ? 'Reply as' : 'Comment as'}
```
=> 
```
{mode === 'reply' ? 'Reply as' : 'Comment as'}
// or
{!isCommenting ? 'Reply as' : 'Comment as'}
```

Miaplacidus replied:
I think ContentType is being specified to make explicit that parentType may match more than a comment or a reply going forward (e.g. a proposal).

mzparacha commented on packages/commonwealth/client/scripts/views/components/Comments/CommentEditor/CommentEditor.tsx:
```diff
@@ -0,0 +1,112 @@
+  userBalance,
+  disabled,
+  cancel,
+  isAdmin,
```
`isAdmin` is used to disable the submit button?

can we update it to be more explicit like `canSubmit`?

```
disabled={disabled && !isAdmin}
```
=> 
```
disabled={disabled && !canSubmit}
```

Miaplacidus replied:
I think these were split apart to indicate that whether a button is disabled is dependent both on the state of the editor (e.g. it's empty) and the user's permissions.

mzparacha commented on packages/commonwealth/client/scripts/views/components/Comments/CommentEditor/CommentEditor.tsx:
```diff
@@ -0,0 +1,112 @@
+      />
+      {tokenPostingThreshold && tokenPostingThreshold.gt(new BN(0)) && (
+        <CWText className="token-req-text">
+          Commenting in {activeTopic?.name} requires{' '}
```
can we pass the topic name directly, since we only use that?

`activeTopic: Topic` => `topicName: string`

Miaplacidus replied:
Done!

mzparacha commented on packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:
```diff
@@ -48,6 +50,9 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
   });
 
+  const onArchivePage =
+    location.pathname === `/${app.activeChainId()}/archived`;
```
should we do `.toLowerCase()`, `/Archived` / `/arChivEd` or any other lettercase combo will break it

Miaplacidus replied:
Testing in the browser, I found that the pathname is downcased regardless of the casing you enter.

mzparacha commented on packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx:
```diff
@@ -32,6 +32,9 @@ type HeaderWithFiltersProps = {
   onIncludeSpamThreads: (includeSpams: boolean) => any;
+  isIncludingArchivedThreads: boolean;
+  onIncludeArchivedThreads: (includeArchived: boolean) => any;
+  onArchivePage?: boolean;
```
`onArchivePage => isPageArchived` or something similar?

Miaplacidus replied:
Done!

mzparacha commented on packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx:
```diff
@@ -32,6 +32,9 @@ type HeaderWithFiltersProps = {
   isIncludingSpamThreads: boolean;
   onIncludeSpamThreads: (includeSpams: boolean) => any;
+  isIncludingArchivedThreads: boolean;
+  onIncludeArchivedThreads: (includeArchived: boolean) => any;
```
`onIncludeArchivedThreads` looks like a bool toggle, can we do `onArchiveChange: (includeArchived: boolean) => any`

Miaplacidus replied:
I feel like that obfuscates the action, which is that the user is trying to include archived threads alongside non-archived threads.

mzparacha commented on packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadOptions/AdminActions/AdminActions.tsx:
```diff
@@ -273,9 +299,10 @@ export const AdminActions = ({
-            ...(hasAdminPermissions ||
-            isThreadAuthor ||
-            (isThreadCollaborator && !thread.readOnly)
+            ...(thread.archivedAt === null &&
```
i feel like `!thread.archivedAt` is safer, curious for your thoughts

Miaplacidus replied:
It should either indicate a date or be `null`, hence the specificity.

mzparacha commented on packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadOptions/ReactionButton/ReactionButton.tsx:
```diff
@@ -119,6 +119,7 @@ export const ReactionButton = ({
         <CWUpvoteSmall
           voteCount={reactors.length}
           disabled={isUserForbidden || disabled}
+          threadArchived={!!thread.archivedAt}
```
`threadArchived => isArchived`

Miaplacidus replied:
Done!

mzparacha commented on packages/commonwealth/client/scripts/views/modals/ArchiveThreadModal.tsx:
```diff
@@ -0,0 +1,87 @@
+
+type ArchiveThreadModalProps = {
+  onModalClose: () => void;
+  thread: Thread;
```
Just to confirm, Is this a presentation component? If not, I feel like `thread: Thread;` is too broad, can we pass specific fields?

Miaplacidus replied:
It's a presentational component.

mzparacha reviewed (commented):
Did the code review (added some small suggestions), Awesome work 🚀, thanks @Miaplacidus. 

Doing the local tests.... Code wise LGTM 🎉.

mzparacha commented:
Just confirming if the cursor on the archive icon needs to be 'pointer'?


https://github.com/hicommonwealth/commonwealth/assets/51641047/5b4bc5cb-32f3-477e-a85e-36a84d59369c

mzparacha commented:
Found some minor things above ^ @Miaplacidus

Miaplacidus commented on packages/commonwealth/client/scripts/views/components/ArchiveTrayWithTooltip.tsx:
Yep yep!
Done.

Miaplacidus commented on packages/commonwealth/client/scripts/views/components/sidebar/discussion_section.tsx:
```diff
@@ -53,6 +54,10 @@ export const DiscussionSection = () => {
     location
   );
+  const matchesArchivedRoute = matchRoutes(
+    [{ path: '/archived' }, { path: ':scope/archived' }],
```
Yep, it works with different casing.

Miaplacidus commented on packages/commonwealth/client/scripts/views/pages/discussions/CommentTree/CommentTree.tsx:
```diff
@@ -437,12 +437,14 @@ export const CommentTree = ({
                 )}
                 <CommentCard
-                  canReply={!!hasJoinedCommunity}
+                  threadArchived={!!thread.archivedAt}
```
Done!

Miaplacidus commented on packages/commonwealth/client/scripts/views/pages/discussions/HeaderWithFilters/HeaderWithFilters.tsx:
```diff
@@ -141,7 +147,11 @@ export const HeaderWithFilters = ({
       <div className="header-row">
         <CWText type="h3" fontWeight="semiBold" className="header-text">
-          {isUndefined(topic) ? 'All Discussions' : topic}
+          {isUndefined(topic)
```
It's weird enough that I'm reluctant to change it. It seems like the question is specifically if topic === undefined as opposed to any other falsy value. 

It is worth further investigation though...

Miaplacidus commented:
1. Not supposed to be a pointer. 
2. Good question! @jessmart1213 How should archived posts appear on the Overview page (I'm guessing the same way they show up on the discussions pages)? Should they appear at all?
3. The actions that are blocked by a thread being archived were previously discussed. If anything has changed on design's side, @jessmart1213, let me know. 
4. Keeping the tags even when a thread is archived is intentional.
5. Same answer as 3. 
6. We keep the comment count, even though you are not allowed to add comments. 
7. We never discussed if all of the options in the cards to the right of the thread on the view thread page(i.e. 'Link Proposal', 'Link Discussion', etc.) should be allowed. @jessmart1213 - any thoughts on this?
8., 9. @zakhap @jessmart1213 How do we handle a thread initially marked as Spam, which is then archived? 
10. This is specified in the design.
11. @zakhap I believe we discussed this and it's intentional. However, it is jarring. We can potentially leave the sorting and filtering off of the Archived page
12. There's no filter being applied. Archive is not a topic. 
13. Please create a bug card for this. Thanks!

zakhap commented:
> 1. Not supposed to be a pointer.

👍 

> 2. Good question! @jessmart1213 How should archived posts appear on the Overview page (I'm guessing the same way they show up on the discussions pages)? Should they appear at all? 

The query for populating the Overview Page should filter out content flagged as either Spam or Archive. This is definitely scope creep and can be created as a fast follow ticket. We should _not_ add new UI here. 

> 3. Not sure, but should we block the pin option if thread is archived?

Yes! It doesn't make much sense to pin a thread to the archive.

> 4. Keeping the tags even when a thread is archived is intentional.

👍  yep

> 5. Should we block the subscribe option if thread is archived?

This should also be disabled 👍, like #3 above. 

> 6. We keep the comment count, even though you are not allowed to add comments.

👍 

> 7. We never discussed if all of the options in the cards to the right of the thread on the view thread page(i.e. 'Link Proposal', 'Link Discussion', etc.) should be allowed. @jessmart1213 - any thoughts on this?

It is fine for this functionality to stay active. This should be further discussed later within product intentions about what it means for a thread to be archived/locked/etc. 

>    8., 9. @zakhap @jessmart1213 How do we handle a thread initially marked as Spam, which is then archived?

It should be marked as both. 

> 8. This is specified in the design.

👍  correct



I will respond to 9-14 shortly

zakhap commented:
> 9. The archived thread, which is also marked as spam, doesn't show up in the "Archived" section

I don't find this behavior to be particularly jarring. The only weird case is on the normal listing page that the user needs to click _both_ spam+archive to see something tagged as both, but I don't think that's a problem. On the archive page, you can simply click "include spam" and it'll show up there.

> 10. Do we need the "include posts flagged as spam" checkbox in the archived section?

Yes! It is fine as is here.

> 11. When we apply any filter from the archived page, it redirects to the discussion page

Def a bug! We should be able to filter for archived threads without seeing all the threads in the topic as well. 

> 12. I think we should show this message in the archived page if there are no archived threads (we do a similar thing on the discussion page)

**I agree. @Miaplacidus this is the only new ask here. Empty state is really important here as all communities will start with an empty Archive page and it will be a new button in their sidebar when the feature is launched.** 

> 13. Unrelated to this PR: the "trending" tag doesn't showup for a thread in the details page

This is a separate bug and I will make a bug ticket for it now!

Miaplacidus commented:
In regards to the bug, @kurtisassad and I have been working to fix it, but it's extremely flakey and appears only under unusual circumstances. Namely, 1. you archive a number of threads in one community 2. you archive yet more threads in another community 3. maybe you see it or maybe you don't. @kurtisassad can't reproduce it at all. In addition, I saw this bug with the spam feature as well under the same circumstances. I'll create a bug ticket around this, but, for now, we're going to disregard the bug.

Miaplacidus commented:
@zakhap 
So, in regards to pinned threads, I have a question/suggestion.
If a thread is currently pinned and an admin decides to archive it, I think we should remove the status of 'pinned' from it. Of course, this means that the admin will have to re-pin the thread, but I think that is the correct level of seriousness archiving should have. It's basically a "dead" thread once it's archived, so I don't think that's inappropriate. 

Please let me know if you have any thoughts or objections! Thanks!

Miaplacidus commented:
@zakhap Also, we do have a message for when there are no archived threads. It is "There are currently no archived threads." I think there is a good argument for making that more generic, since it's not entirely accurate when you are filtering/sorting by newest, stage, etc. Something more like "There are no archived threads matching your filter." would fit all situations.

mzparacha reviewed (approved):
This looks great. Thanks for all the changes @Miaplacidus 🙏.
